articles/introduction-to-built-in-data-structures-in-java/index.md

```diff
@@ -0,0 +1,304 @@
+You will also learn how to work with built-in data structures.
+
+### Table of contents
+
+- [Introduction](#introduction)
+- [Data Structures](#data-structures)
+- [Conclusion](#conclusion)
+- [Further Reading](#further-reading)
+
+### Introduction
+
+> According to Wikipedia, a data structure is a data organization, management, and storage format that enables efficient access and modification. More precisely, a data structure is a collection of data values, the relationships among them, and the functions or operations that can be applied to the data.
+
+#### Built-in vs. User-defined Data Structures
+
+Data structures like Stack, or Queue can be created on our own using other basic data structures. For example, Stacks can be created using Arrays or Linked lists. These are known as User-defined data structures.
+
+While built-in data structures do not require the creation of a data structure from scratch. But, we can make use of prebuilt methods that abstract the working of the data structure.
+
+By using built-in data structures, we can improve productivity by writing the operations of data structures from scratch. Also, built-in data structures are generally written with the most efficient time and space complexities.
+
+Time and space complexities are measures of code that ensures the effective usage of time and memory while running a code snippet. By reducing these complexities, the efficiency of data structures can be improved.
+
+### Data Structures
+
+Now, let's have a look at a few built-in data structures, and we'll see how to work with them.
+
+#### ArrayList
+
+In the `java.util` package, `ArrayList` is a built-in data structure very similar to `Array`.
+
+The difference between `Array` and `ArrayList` is the size. In `Array`, the length is fixed, so the memory allocated for storing values is fixed. Whereas, in `ArrayList`, the list is resizable, which allows in choosing of dynamic size. Here, the memory is created dynamically.
+
+Syntax for creating ArrayList:
+
+```java
+import java.util.ArrayList; // Import the ArrayList class
+
+ArrayList<String> list1 = new ArrayList<String>(); // Create an ArrayList object
+```
+
+##### Add Elements
+
+When using an `Array`, we have to manually assign the values or variables to the particular index. But, in `ArrayList` adding elements to the list is made simpler with the use of the `add()` method.
+
+Example:
+
+```java
+list1.add("Hello"); // Add "Hello" to the ArrayList
+// OUTPUT: list1 = ["Hello"]
+list1.add("World"); // Add "World" to the ArrayList
+// OUTPUT: list1 = ["Hello", "World"]
+```
+
+##### Update Elements
+
+Updation of elements in `Array` is similar to addition, where the index of the array is found and replaced with the desired value.
+
+In `ArrayList`, we use the `set(index, value)` method to replace the element `value` at a particular `index`.
+
+Example:
+
+```java
+list1.set(1, "Welcome"); // Replaces "World" with "Welcome" in list1
+// OUTPUT: list1 = ["Hello", "Welcome"]
```

Comment:
Hi @cyberShaw . Thanks for the suggestion.

As you mentioned, adding output statements using `System.out.println()` is not possible for certain data structures, the data structure must be iterated using a for loop or any other iterator. In that case, should I write a function for iterating, and call it everytime for displaying the output?

Reply:
Yeah, it would be great if you can do that. Beginners could easily just copy and paste the code to see the results and easily learn how the data structure works